src/Features/LanguageServer/Microsoft.CommonLanguageServerProtocol.Framework/Protocol/Internal/Text/ClassifiedTextElement.cs

```diff
@@ -0,0 +1,41 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using Roslyn.Text.Adornments;
+
+namespace Roslyn.Text.Adornments
```

Comment:
I would *really* like to avoid redefining the VS types if at all possible.  Today using the protocol libs as-is, we can make the protocol project only depend on the public definitions by default (add more abstraction around the creation of LSP types).

However if we're using our own definitions I can see using the actual VS LSP library definitions could be hard, especially if we want a `VSHover` to be a subclass of an LSP `Hover`.  The only workaround I can think of is `object` types, but that doesn't seem super nice. 

Re-defining the vs only types is going to be annoying especially since that is where a lot of the changes go (so keeping in syc could be challenging).